### PR TITLE
[feature-wip](multi-catalog)add properties converter

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Refactor properties of each cloud ,  use property converter to convert properties  accessing fe metadata and be data.
user docs https://github.com/apache/doris/pull/18287

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        for (Map.Entry<String, String> entry : props.entrySet()) {
+            if (entry.getKey().startsWith(ObsProperties.OBS_FS_PREFIX)) {
+                obsProperties.put(entry.getKey(), entry.getValue());
+            }
+        }
+        BeAWSProperties.setBeAWSPropertiesFromObs(obsProperties);
+        return obsProperties;
+    }
+
+    private static Map<String, String> convertToS3EnvProperties(Map<String, String> properties,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:109]:
```suggestion
    private static Map<String, String> convertToS3HadoopFSProperties(Map<String, String> properties,
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+                || props.containsKey(DataLakeConfig.CATALOG_ENDPOINT)) {
+            return convertToDLFProperties(props, DLFProperties.getCredential(props));
+        }
+        return props;
+    }
+
+    /**
+     * Convert properties defined at doris to FE S3 client properties and BE S3 client properties
+     */
+    public static Map<String, String> storageConvert(Map<String, String> props) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:72]:
```suggestion
    public static Map<String, String> convertToHadoopFSProperties(Map<String, String> props) {
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+package org.apache.doris.datasource.property;
+
+import org.apache.doris.datasource.property.constants.S3Properties;
+
+import org.apache.hadoop.fs.obs.OBSConstants;
+import org.apache.hadoop.fs.s3a.Constants;
+
+import java.util.Map;
+
+public class BeAWSProperties {
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java:28]: BeS3ClientProperties

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        properties.put(S3Properties.Environment.ENDPOINT, endpoint);
+        String region = PropertyConverter.getRegionOfEndpoint(endpoint);
+        properties.put(S3Properties.Environment.REGION, region);
+        if (properties.containsKey(OBSConstants.ACCESS_KEY)) {
+            properties.put(S3Properties.Environment.ACCESS_KEY, properties.get(OBSConstants.ACCESS_KEY));
+        }
+        if (properties.containsKey(OBSConstants.SECRET_KEY)) {
+            properties.put(S3Properties.Environment.SECRET_KEY, properties.get(OBSConstants.SECRET_KEY));
+        }
+        if (properties.containsKey("fs.obs.session.token")) {
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java:72]: the OBSConstants bind to "fs.obs.credentials.provider" in huawei obs client

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java (hunk 1 above)
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:110]: S3HadoopFSProperties like this: "fs.s3a.xxx"
So named the function  as S3EnvProperties to distinguish it from S3HadoopFSProperties

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/property/BeAWSProperties.java]: define it in ObsProperties as our properties

File: fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java [hunk 5]
```diff
@@ -31,26 +34,27 @@
...
+        props = PropertyConverter.convertToMetaProperties(props);
         catalogProperty = new CatalogProperty(resource, props);
     }
 
     @Override
     protected void initLocalObjectsImpl() {
         icebergCatalogType = ICEBERG_GLUE;
         GlueCatalog glueCatalog = new GlueCatalog();
-        // AWSGlueAsync glueClient;
         Configuration conf = setGlueProperties(getConfiguration());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java:50]: `setGlueProperties()` is an empty method

File: fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java [hunk 6]
```diff
@@ -31,26 +34,27 @@
...
-        String metastoreUris = catalogProperty.getOrDefault(CatalogProperties.WAREHOUSE_LOCATION, "");
-        if (StringUtils.isEmpty(metastoreUris)) {
-            throw new IllegalArgumentException("Missing glue properties 'warehouse'.");
-        }
-        catalogProperties.put(CatalogProperties.WAREHOUSE_LOCATION, metastoreUris);
+        Map<String, String> catalogProperties = catalogProperty.getHadoopProperties();
+        String warehouse = catalogProperty.getOrDefault(CatalogProperties.WAREHOUSE_LOCATION, CHECKED_WAREHOUSE);
+        catalogProperties.put(CatalogProperties.WAREHOUSE_LOCATION, warehouse);
+        catalogProperties.put(AwsProperties.S3FILEIO_ENDPOINT,
+                catalogProperties.getOrDefault(Constants.ENDPOINT, conf.get(S3Properties.Env.ENDPOINT)));
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java:57]: Why the key name in `conf` is `S3Properties.Env.ENDPOINT`?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.doris.datasource.property.constants.ObsProperties;
+import org.apache.doris.datasource.property.constants.OssProperties;
+import org.apache.doris.datasource.property.constants.S3Properties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class S3ClientBEProperties {
+
+    public static Map<String, String> getBeFSProperties(Map<String, String> properties) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java:30]: Add comment to explain the method, better give an example.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java (hunk 6 above)
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergGlueExternalCatalog.java:57]: S3Properties.Env.ENDPOINT is BE endpoint.
set AwsProperties.S3FILEIO_ENDPOINT by converted s3 endpoint or by BE s3 endpoint default.

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: docs/zh-CN/docs/lakehouse/multi-catalog/hive.md [hunk 8]
```diff
@@ -114,23 +116,85 @@ CREATE CATALOG hive PROPERTIES (
...
+
+元数据存储在Glue，示例如下：
+
+```sql
+CREATE CATALOG hive PROPERTIES (
+    "type"="hms",
+    "hive.metastore.type" = "glue",
+    "glue.endpoint" = "https://glue.us-east-1.amazonaws.com",
+    "glue.access-key" = "ak",
+    "glue.secret-key" = "sk"
```
[the remark this answers is not in the document]
Comment (wsjz) [reply docs/zh-CN/docs/lakehouse/multi-catalog/hive.md]: https://github.com/apache/doris/pull/18287

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.49 seconds
 stream load tsv:          457 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230331133638_clickbench_pr_123468.html

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run bulidall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run clickbench

**Comment by wsjz** [comment]
run p1

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run clickbench

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**